models/issues/milestone.go

```diff
@@ -82,9 +84,18 @@ func (m *Milestone) BeforeUpdate() {
 
 // AfterLoad is invoked from XORM after setting the value of a field of
 // this object.
-func (m *Milestone) AfterLoad() {
+func (m *Milestone) AfterLoad(session *xorm.Session) {
 	m.NumOpenIssues = m.NumIssues - m.NumClosedIssues
-	if m.DeadlineUnix.Year() == 9999 {
+	if m.DeadlineUnix == 0 {
+		return
+	}
+	// for legacy reasons, all years after 9000 are considered as no deadline
+	if m.DeadlineUnix.Year() > 9000 {
+		m.DeadlineUnix = 0
+		m.IsOverdue = false
+		if _, err := session.ID(m.ID).NoAutoTime().Cols("deadline_unix", "is_overdue").Update(m); err != nil {
+			log.Error("AfterLoad update legacy deadline failed: %v", err)
+		}
```

Comment:
Why not add a migration to convert all legacy deadlines to the new value in the DB instead of keeping this code?

Reply:
To make the backport easier.